examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm

```diff
@@ -0,0 +1,142 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "SetupPayloadParseCommand.h"
+#import <CHIP/CHIP.h>
+#import <CHIP/CHIPError_Internal.h>
+
+using namespace ::chip;
+
+namespace {
+
+#if CHIP_PROGRESS_LOGGING
+
+NSString * CustomFlowString(CHIPCommissioningFlow flow)
+{
+    switch (flow) {
+    case kCommissioningFlowStandard:
+        return @"STANDARD";
+    case kCommissioningFlowUserActionRequired:
+        return @"USER ACTION REQUIRED";
+    case kCommissioningFlowCustom:
+        return @"CUSTOM";
+    case kCommissioningFlowInvalid:
+        return @"INVALID";
+    }
+
+    return @"???";
+}
+
+#endif // CHIP_PROGRESS_LOGGING
+
+} // namespace
+
+void SetupPayloadParseCommand::LogNSError(const char * logString, NSError * error)
+{
+    CHIP_ERROR err = [CHIPError errorToCHIPErrorCode:error];
+    if (err == CHIP_NO_ERROR) {
+        ChipLogProgress(chipTool, "%s: %s", logString, chip::ErrorStr(err));
+    } else {
+        ChipLogError(chipTool, "%s: %s", logString, chip::ErrorStr(err));
+    }
+}
+
+CHIP_ERROR SetupPayloadParseCommand::Run()
+{
+    NSString * codeString = [NSString stringWithCString:mCode encoding:NSASCIIStringEncoding];
+    NSError * error;
+    CHIPSetupPayload * payload;
+    CHIPOnboardingPayloadType codeType;
+    if (IsQRCode(codeString)) {
```

Comment:
As a followup: `CHIPOnboardingPayloadParser` should just do that check itself....

Reply:
Ok. Created a new [issue](https://github.com/project-chip/connectedhomeip/issues/20311).